linera-views/tests/map_view_tests.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use linera_views::{
+    common::get_interval,
+    map_view::ByteMapView,
+    memory::create_memory_context,
+    views::{CryptoHashRootView, RootView, View},
+};
+use rand::{Rng, SeedableRng};
+use std::collections::BTreeMap;
+
+#[derive(CryptoHashRootView)]
+pub struct StateView<C> {
+    pub map: ByteMapView<C, u8>,
+}
+
+fn remove_by_prefix<V>(map: &mut BTreeMap<Vec<u8>, V>, key_prefix: Vec<u8>) {
+    let key_list = map
+        .range(get_interval(key_prefix.clone()))
+        .map(|x| x.0.to_vec())
+        .collect::<Vec<_>>();
+    for key in key_list {
+        map.remove(&key);
+    }
```

Comment:
For the tests we could just use `BTreeMap::retain`, so the "expected" values don't rely on `get_interval`.

Reply:
Thanks, that is indeed better.
Now, I think it is better in general. Why not do this for the non-test code?

Reply:
I think `range(interval)` will be faster than `retain` for large collections, because it doesn't iterate over all the entries.

Actually, for _very_ large collections, applying `split_off` twice might be the fastest option, since it doesn't even iterate over all the keys _in_ the interval… but I'm not sure how that's implemented internally.

Reply:
Ok, so let's leave it here.